shell/components/form/KeyValue.vue

```diff
@@ -662,39 +672,51 @@ export default {
               {{ binaryTextSize(row.value) }}
             </div>
-            <CodeMirror
-              v-else-if="valueMarkdownMultiline"
-              ref="cm"
-              data-testid="code-mirror-multiline-field"
-              :class="{['focus']: codeMirrorFocus[i]}"
-              :value="row[valueName]"
-              :as-text-area="true"
-              :mode="mode"
-              @onInput="onInputMarkdownMultiline(i, $event)"
-              @onFocus="onFocusMarkdownMultiline(i, $event)"
-            />
-            <TextAreaAutoGrow
-              v-else-if="valueMultiline"
-              v-model="row[valueName]"
-              :class="{'conceal': valueConcealed}"
-              :disabled="disabled || isProtected(row.key)"
-              :mode="mode"
-              :placeholder="valuePlaceholder"
-              :min-height="40"
-              :spellcheck="false"
-              @input="queueUpdate"
-            />
-            <input
+            <div
               v-else
-              v-model="row[valueName]"
-              :disabled="isView || disabled || isProtected(row.key)"
-              :type="valueConcealed ? 'password' : 'text'"
-              :placeholder="valuePlaceholder"
-              autocorrect="off"
-              autocapitalize="off"
-              spellcheck="false"
-              @input="queueUpdate"
+              class="value-container"
+              :class="{ 'upload-button': parseValueFromFile }"
             >
+              <CodeMirror
+                v-if="valueMarkdownMultiline"
+                ref="cm"
+                data-testid="code-mirror-multiline-field"
+                :class="{['focus']: codeMirrorFocus[i]}"
+                :value="row[valueName]"
+                :as-text-area="true"
+                :mode="mode"
+                @onInput="onInputMarkdownMultiline(i, $event)"
+                @onFocus="onFocusMarkdownMultiline(i, $event)"
+              />
+              <TextAreaAutoGrow
+                v-else-if="valueMultiline"
+                v-model="row[valueName]"
+                :class="{'conceal': valueConcealed}"
+                :disabled="disabled || isProtected(row.key)"
+                :mode="mode"
+                :placeholder="valuePlaceholder"
+                :min-height="40"
+                :spellcheck="false"
+                @input="queueUpdate"
+              />
+              <input
+                v-else
+                v-model="row[valueName]"
+                :disabled="isView || disabled || isProtected(row.key)"
+                :type="valueConcealed ? 'password' : 'text'"
+                :placeholder="valuePlaceholder"
+                autocorrect="off"
+                autocapitalize="off"
+                spellcheck="false"
+                @input="queueUpdate"
+              >
+              <FileSelector
+                v-if="parseValueFromFile && readAllowed && !isView && row[keyName] && isValueFieldEmpty(row[valueName])"
```

Comment:
I had a play around and I'm not sure we need to hide the upload button if there's no name. Hiding the `Upload` when there's content makes a bit more sense (just from the perspective of how the button is positioned and avoiding overlap).

We should raise how this behaves in the UI Demo on Thursday